### PR TITLE
Add a CMake external build for c-ares

### PR DESCRIPTION
**Pull request by wilhuff**

### CODE REVIEW

File: cmake/external/cares.cmake [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+include(ExternalProject)
+
+ExternalProject_Add(
+  cares
```
Comment (rsgowman) [inline cmake/external/cares.cmake]: I don't know how much you *care*, but c-ares is more often stylized with a dash. ("c-ares" vs "cares").

File: cmake/external/cares.cmake [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  PREFIX ${PROJECT_BINARY_DIR}/external/cares
+
+  CMAKE_ARGS
+    -DCMAKE_BUILD_TYPE:STRING=${CMAKE_BUILD_TYPE}
+    -DCMAKE_INSTALL_PREFIX:STRING=${FIREBASE_INSTALL_DIR}
+    -DCARES_STATIC:BOOL=ON
+    -DCARES_SHARED:BOOL=OFF
+    -DCARES_STATIC_PIC:BOOL=ON
+
+  UPDATE_COMMAND ""
```
Comment (rsgowman) [inline cmake/external/cares.cmake]: I suspect (but haven't verified) that we can drop `UPDATE_COMMAND ""` for all cases except where we fetch from source control (such as grpc).

File: cmake/external/cares.cmake (hunk 1 above)
Comment (wilhuff) [reply cmake/external/cares.cmake]: I almost named this branch "wilhuff/cares". It turns out, I do!

Renamed the target and the file.

File: cmake/external/cares.cmake (hunk 2 above)
Comment (wilhuff) [reply cmake/external/cares.cmake]: Verified and removed.

**Pull request merged**